### DISCUSSION TITLE
Howto: send windowPost while resilver (zfs pool) for small/midsize miners

### DISCUSSION THREAD

**Discussion by Reiers** [filecoin-project/lotus#7522] [Tutorials]
### Resilver ZFS pools and show windowPosts

One faulty HDD or two - can be a headache especially when the resilver starts right after you have replace it/them.

**Resilver requires a lot of IO from your storage server and does not leave room for windowPost.**
Most likely the partition will take to long to load and it will not be able to send/prove windowPost in time.

You cant stop the resilver - but you can slow the process down enough to still send windowPost. 

First the default config for ZFS Pools, remember to adjust it back as soon as the windowPosts are done/sent:

```
zfs_resilver_min_time_ms = 3000
zfs_scan_vdev_limit = 4194304
```

**Check your system, if they are not default, SAVE the output/print of the cmds under so you know what to revert the values back to:**
```
cat /sys/module/zfs/parameters/zfs_resilver_min_time_ms
cat /sys/module/zfs/parameters/zfs_scan_vdev_limit
```

**Here is how to slow down and revert back:**

Slow down:
```
echo 500 > /sys/module/zfs/parameters/zfs_resilver_min_time_ms
echo 1000 > /sys/module/zfs/parameters/zfs_scan_vdev_limit
```

Revert back to default speed:
```
echo 3000 > /sys/module/zfs/parameters/zfs_resilver_min_time_ms
echo 4194304 > /sys/module/zfs/parameters/zfs_scan_vdev_limit
```

**Tested on my own storage servers. NOT all servers are created equally - use at your own risk !**

**Comment by stuberman** [comment]
This is a keeper!
When you change the settings do you need to run sudo?

**Reply by Reiers** [reply]
Correct - sudo before or be logged in as root do to these changes !

**Reply by Angelo-gh3990** [reply]
Nice one, thanks @Reiers
